hw/ip/pwrmgr/doc/_index.md

```diff
@@ -303,7 +304,55 @@ There is currently no standardized way to do this, so it is up to the debugging
 
 # Programmers Guide
 
-Please see [power management programmers model](https://docs.google.com/document/d/1w86rmvylJgZVmmQ6Q1YBcCp2VFctkQT3zJ408SJMLPE/edit?usp=sharing) for now.
+The process in which the power manager is used is highly dependent on the system's topology.
+The following proposes one method for how this can be done.
+
+Assume first the system has the power states described [above](#supported-low-power-modes).
+
+## Programmer Sequence for Entering Low Power
+
+1. Disable interrupts
+2. Enable desired wakeup and reset sources in {{< regref "WAKEUP_EN" >}} and {{< regref "RESET_EN" >}}.
+3. Perform any system-specific low power entry steps, e.g.
+   - Interrupt checks (if something became pending prior to disable)
+4. Configure low power mode in {{< regref "CONTROL" >}}.
+5. Set low power hint in {{< regref "LOW_POWER_HINT" >}}.
+6. Execute wait-for-interrupt instruction on the processing host.
+
+### Possible Exits
+
+Once low power is entered, the system may exit due to several reasons.
+1. Graceful low power exit - This exit occurs when some source in the system gracefully wakes up the power manager.
+2. [Fall through exit](#fall-through-handling) - This exit occurs when an interrupt manages to break the wait-for-interrupt loop.
+3. [Aborted entry](#abort-handling) - This exit occurs when low power entry is attempted with an ongoing non-volatile transaction.
+
+## Programmer Sequence for Exiting Low Power
+
+There are two separate cases for low power exit.
+One is exiting from deep sleep, and the other is exiting from normal sleep.
+
+### Exiting from Deep Sleep
+
+When exiting from deep sleep, the system begins execution in ROM.
+
+1. Complete normal preparation steps.
+2. Check reset cause in [rstmgr]({{< relref "hw/ip/rstmgr/doc" >}})
+3. Re-enable modules that have powered down.
+4. Disable wakeup recording through {{< regref "WAKE_INFO_CAPTURE_DIS" >}}.
+5. Check which source woke up the system through {{< regref "WAKE_INFO" >}}.
+6. Take appropriate steps to handle the wake and resume normal operation.
+
+### Exiting from Normal Sleep
+
+The handling for fall-through and abort are similar to normal sleep exit.
+Since in these scenarios the system was not reset, software continues executing the instruction after the wait-for-interrupt invocation.
+
+1. Check exit condition to determine appropriate steps.
+2. Clear low power hints and configuration in {{< regref "CONTROL" >}}.
+3. Disable wakeup sources and stop recording.
+4. Re-enable interrupts for normal operation.
+
+For an in-depth discussion, please see [power management programmers model](https://docs.google.com/document/d/1w86rmvylJgZVmmQ6Q1YBcCp2VFctkQT3zJ408SJMLPE/edit?usp=sharing) for additional details.
```

Comment:
Is this actually publicly visible? If not, I am not sure whether we should link it publicly...

Reply:
i think eventually it will be made public.